game-core/src/main/java/games/strategy/engine/framework/GameDataUtils.java

```diff
@@ -28,6 +28,20 @@ public static GameData cloneGameDataWithoutHistory(
     return dataCopy;
   }
 
+  public static byte[] serializeGameDataWithoutHistory(final GameData data) {
```

Comment:
This is an interesting way to reduce the game data serialization size. I've been thinking we'll want to eventually have a param object for calculations that has a copy of *just* the data it needs. In the meantime this is interesting to move us in that direction.

Reply:
In an ideal world this "copy" process would be super cheap to do, so that we don't have to roll back changes manually after each simulation run, but could just create a new copy instead.

The benefit of this intermediate solution is that there's absolutely no locking involved once we have the byte array.
This way all threads can create their own instance of the GameData independent of each other.